### DISCUSSION TITLE
Fonts preload

### DISCUSSION THREAD

**Discussion by ilyaDegtyarenko** [nuxt/nuxt#17389] [Questions]
How can I preload fonts?
When I do this in **css config block**, they jump when the page reloads

```
// https://v3.nuxtjs.org/api/configuration/nuxt.config
export default defineNuxtConfig({
    css: [
        '~/assets/styles/css/font.css',
        ...
    ],
    ...
})

```

**Comment by Flanker72** [comment]
You may use `app.head.link` array to pass the preload and preconnect content like this:

```
import { defineNuxtConfig } from "nuxt";

export default defineNuxtConfig({
  app: {
    head: {
      link: [
        { rel: "preconnect", href: "https://fonts.googleapis.com" },
        { rel: "preconnect", href: "https://fonts.gstatic.com", crossorigin: true },
        { rel: "preload", href: "_nuxt/assets/fonts/MonofontoRegular.ttf", as: "font" },
      ],
    },
  },
});
```

However if you need to preload locally served fonts from the `assets` folder - it won't be resolved automatically (in the example above I use the direct link to the generated assets).

**Reply by ilyaDegtyarenko** [reply]
I followed your example, but got an error
`GET http://localhost:3000/page/_nuxt/assets/fonts/MyFont-Medium.woff net::ERR_ABORTED 404 (Not Found)`
**The path to the fonts is correct**

```
export default defineNuxtConfig({
    app: {
        head: {
            link: [
                { rel: "preload", href: "_nuxt/assets/fonts/MyFont-Medium.woff", as: "font" },
                { rel: "preload", href: "_nuxt/assets/fonts/MyFont-Medium.woff2", as: "font" }
            ],
        }
    }
})
```

**Reply by Flanker72** [reply]
My bad, missed the leading slash. However, in the production mode assets will be renamed/chunked. You may put the fonts inside `public` folder, and then use the preload as expected (as a static file from the root of your project).
Assume you have `public/fonts` folder, then the links should be:
```
link: [
    { rel: "preload", href: "/fonts/MyFont-Medium.woff", as: "font" },
    { rel: "preload", href: "/fonts/MyFont-Medium.woff2", as: "font" }
],
```

**Reply by ilyaDegtyarenko** [reply]
This helped, but did not completely solve the problem (
Can you explain to me what I'm doing wrong?

Now I get warning in the console:
`The resource http://localhost:3000/fonts/MyFont-Medium.woff2 was preloaded using link preload but not used within a few seconds from the window's load event. Please make sure it has an appropriate `as` value and it is preloaded intentionally.
...`

**./nuxt.config.ts**
```
export default defineNuxtConfig({
    ...,
    app: {
        head: {
            link: [
                {rel: "preload", href: "/fonts/MyFont-Medium.woff", as: "font"},
                {rel: "preload", href: "/fonts/MyFont-Medium.woff2", as: "font"}
            ]
        }
    }
})
```

I tried to connect fonts in two ways:

**First in ./nuxt.config.ts**
```
export default defineNuxtConfig({
    css: [
        '~/public/css/font.css',
    ],
})
```
**Or second in ./app.vue**
```
<style>
    @import "public/css/font.css";
</style>
```

**public/css/font.css**
```
@font-face {
    font-family: 'MyFont';
    src: url('/fonts/MyFont-Regular.woff2') format('woff2'), url('/fonts/MyFont-Regular.woff') format('woff');
    ...
}

@font-face {
    font-family: 'MyFont';
    src: url('/fonts/MyFont-Medium.woff2') format('woff2'), url('/fonts/MyFont-Medium.woff') format('woff');
    ...
}
```

**When the pages are refreshed, the fonts still jump. Once or twice in several reloads. Or is it normal?**
https://user-images.githubusercontent.com/37239202/175558135-1761094f-43ff-49cf-87d2-2849ae8b22e6.mov

**Reply by ilyaDegtyarenko** [reply]
🎉 Resolved as follows:
```
export default defineNuxtConfig({
    app: {
            link: [
                {
                    rel: 'preload',
                    href: '/fonts/MyFont-Medium.woff',
                    type: 'font/woff',
                    as: 'font',
                    importance: 'high',
                    crossorigin: 'anonymous'
                },
                {
                    rel: 'preload',
                    href: '/fonts/MyFont-Medium.woff2',
                    type: 'font/woff2',
                    as: 'font',
                    importance: 'high',
                    crossorigin: 'anonymous'
                },
                {
                    rel: 'stylesheet',
                    href: '/css/font.css',
                    type: 'text/css',
                    importance: 'high',
                    crossorigin: 'anonymous'
                }
            ]
        }
    }
})

```

@Flanker72 Thanks for the tips 👍
